src/help.ts

```diff
@@ -42,14 +47,74 @@ export abstract class BBHelpClient {
   }
 
   public static openWidgetToHelpKey(helpKey: string = BBHelpClient.currentHelpKey): void {
-    BBHELP.HelpWidget.open(helpKey);
+    BBHelpClient.executeWhenReady(() => {
+        BBHELP.HelpWidget.open(helpKey);
+      });
   }
 
   public static getCurrentHelpKey(): string {
     return BBHelpClient.currentHelpKey || BBHelpClient.defaultHelpKey;
   }
 
   public static toggleOpen(): void {
-    BBHELP.HelpWidget.toggleOpen();
+    BBHelpClient.executeWhenReady(() => {
+      BBHELP.HelpWidget.toggleOpen();
+    });
+  }
+
+  public static openWidget(): void {
+    BBHelpClient.executeWhenReady(() => {
+      BBHELP.HelpWidget.open();
+    });
+  }
+
+  public static closeWidget(): void {
+    BBHelpClient.executeWhenReady(() => {
+      BBHELP.HelpWidget.close();
+    });
+  }
+
+  public static disableWidget(): void {
+    BBHelpClient.executeWhenReady(() => {
+      BBHELP.HelpWidget.disableWidget();
+    });
+  }
+
+  public static enableWidget(): void {
+    BBHelpClient.executeWhenReady(() => {
+      BBHELP.HelpWidget.enableWidget();
+    });
+  }
+
+  public static executeWhenReady(actionCallback: () => void): Promise<any> {
+    return BBHelpClient.ready()
+      .then(() => {
+        return BBHELP.HelpWidget.ready();
+      })
+      .then(() => {
+        actionCallback();
+      })
+      .catch((err: string) => {
+        console.error(err);
+      });
+  }
+
+  public static ready(): Promise<any> {
```

Comment:
You'll have two "ready" functions that are publicly accessible. Did it look awful having the `ready()` logic buried in the `executeWhenReady()` function?

Reply:
The problem here is we need to check both that the HelpWidget is ready, and the client itself is ready. Otherwise it was trying to make calls such as disableWidget once the client was ready, but the help widget wasn't fully created yet.

Reply:
I would also argue that the two serve different purposes.  One is a ready check, to see if things are good to go, the other executes a callback once it clears the ready checks.